### DISCUSSION TITLE
Editing CSS After Import / While Editing Template

### DISCUSSION THREAD

**Discussion by drobin04** [GrapesJS/grapesjs#5540] [Ideas]
It feels odd to me, as someone who's found this app, that I can import HTML/CSS to a page, and that I can export HTML and CSS out, and that I can add classes to items on the page... But can't add CSS rules while I'm editing the page. 

It's possible that I'm missing something, but i've looked. I tried looking at the docs. I don't see an option to edit the CSS anywhere. Are we meant to just export out and do that off the page? It feels like if this were here, it would make this infinitely more useful.

**Comment by artf** [comment]
Not sure how exactly you would expect to edit the CSS as it depends on how much control you want to give to the end users (eg. single rule, all rules) but you can definitely update the CSS created inside the editor ([CSS Composer](https://grapesjs.com/docs/api/css_composer.html) keeps track of all the CSS rules).

In [GrapesJS Studio](https://studio.grapesjs.com/), for example, it's possible to list all the rules and edit their CSS via code editor.
<img width="794" alt="edit-css" src="https://github.com/GrapesJS/grapesjs/assets/11614725/3fed39a7-3896-42a6-a93f-522944b4e7f1">

**Reply by drobin04** [reply]
That style catalog bit is huge. I never knew that existed.... it took a lot of poking around today for me to find the button that opens that in studio, since that button doesn't exist on the demo webpage editor.

If this style catalog feature were in the demo editor there, that would be huge. 

The other vision I had was fairly brute-ish... I'd just like to be able to enter a CSS rule, one at a time as I'm editing. It seems you can do that in the studio via that style catalog UI by editing the CSS for your classes, but as far as I can see there's not a way to accomplish that on the webpage demo editor.

**Reply by artf** [reply]
My point here, GrapesJS is a framework not an end-user application, so the API is here. How your editor should allow editing CSS is a product question.
